build_tools/waf_dynamo.py

```diff
@@ -1508,12 +1546,36 @@ def detect(conf):
 
     platform_setup_tools(conf, build_util)
 
-    conf.check_tool('compiler_cc')
-    conf.check_tool('compiler_cxx')
+    # jg: this whole thing is a 'dirty hack' to be able to pick up our own SDKs
+    if 'win32' in platform:
+        includes = sdkinfo['includes']['path']
+        libdirs = sdkinfo['lib_paths']['path']
+        bindirs = sdkinfo['bin_paths']['path']
+
+        bindirs.append(build_util.get_binary_path())
+        bindirs.append(build_util.get_dynamo_ext_bin())
+
+        # The JDK dir doesn't get added since we use no_autodetect
+        bindirs.append(os.path.join(os.getenv('JAVA_HOME'), 'bin'))
+
+        # there's no lib prefix anymore so we need to set our our lib dir first so we don't
+        # pick up the wrong hid.lib from the windows sdk
+        libdirs.insert(0, build_util.get_dynamo_home('lib', build_util.get_target_platform()))
+
+        conf.env['PATH']     = bindirs + sys.path + conf.env['PATH']
+        conf.env['INCLUDES'] = includes
+        conf.env['LIBPATH']  = libdirs
+        conf.load('msvc', funs='no_autodetect')
+
+        if not Options.options.skip_codesign:
+            conf.find_program('signtool', var='SIGNTOOL', mandatory = True, path_list = bindirs)
```

Comment:
This is an unfortunate change in waf, if we don't specify our own toolchain like this waf will pick the one from the windows registry which might be a completely different version than our SDK which means linking with wrong runtime on CI. This way is recommended by waf now if we get the wrong version..